arches_component_lab/src/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor.vue

```diff
@@ -0,0 +1,199 @@
+<script setup lang="ts">
+import { onMounted, useTemplateRef, ref, watch, computed } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Message from "primevue/message";
+import FileUpload, {
+    type FileUploadRemoveEvent,
+    type FileUploadSelectEvent,
+} from "primevue/fileupload";
+import Image from "primevue/image";
+import Button from "primevue/button";
+import { FormField } from "@primevue/forms";
+
+import type { FormFieldResolverOptions } from "@primevue/forms";
+import type { FileReference } from "@/arches_component_lab/widgets/types.ts";
+const { $gettext } = useGettext();
+const allowedFileTypes = ref();
+const props = defineProps<{
+    initialValue: FileReference[] | null | undefined;
+    graphSlug: string;
+    nodeAlias: string;
+    widgetData: {
+        config: {
+            acceptedFiles: string;
+            maxFilesize: number;
+            rerender: boolean;
+            label: string;
+        };
+    };
+    nodeData: {
+        config: {
+            imagesOnly: boolean;
+            maxFiles: number;
+            maxFileSize: number;
+        };
+    };
+}>();
+
+const formFieldRef = useTemplateRef("formFieldRef");
+const currentValues = ref<FileReference[]>();
+
+watch(
+    () => props.initialValue,
+    (newValue) => {
+        if (newValue) {
+            currentValues.value = newValue;
+        } else {
+            currentValues.value = [];
+        }
+    },
+);
+
+onMounted(() => {
+    allowedFileTypes.value = props.nodeData.config.imagesOnly ? "image/*" : "*";
+
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        newFiles: [],
+        deletedFiles: [],
+    };
+
+    if (props.initialValue) {
+        currentValues.value = props.initialValue;
+    }
+});
+
+const currentMax = computed(() => {
+    if (currentValues.value) {
+        return (
+            props.nodeData.config.maxFiles - (currentValues.value.length ?? 0)
+        );
+    } else {
+        return props.nodeData.config.maxFiles;
+    }
+});
+
+function resolver(e: FormFieldResolverOptions) {
+    validate(e);
+}
+
+function select(event: FileUploadSelectEvent) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        newFiles: event.files,
+    };
+}
+
+function remove(event: FileUploadRemoveEvent) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        newFiles: event.files,
+    };
+}
```

Comment:
I think in a recent iteration we've moved away from this pattern to set values in the form, opting to use `resolver` instead. Is it possible to use that pattern here?